environment/src/test/java/com/sequenceiq/environment/environment/encryption/EnvironmentEncryptionServiceTest.java

```diff
@@ -0,0 +1,139 @@
+package com.sequenceiq.environment.environment.encryption;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.HashMap;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.sequenceiq.cloudbreak.cloud.CloudConnector;
+import com.sequenceiq.cloudbreak.cloud.EncryptionResources;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudPlatformVariant;
+import com.sequenceiq.cloudbreak.cloud.model.Region;
+import com.sequenceiq.cloudbreak.cloud.model.encryption.DiskEncryptionSetCreationRequest;
+import com.sequenceiq.environment.credential.domain.Credential;
+import com.sequenceiq.environment.credential.v1.converter.CredentialToCloudCredentialConverter;
+import com.sequenceiq.environment.environment.domain.Environment;
+import com.sequenceiq.environment.environment.domain.EnvironmentTags;
+import com.sequenceiq.environment.environment.dto.EnvironmentDto;
+import com.sequenceiq.environment.environment.dto.LocationDto;
+import com.sequenceiq.environment.environment.service.EnvironmentTagProvider;
+import com.sequenceiq.environment.parameter.dto.AzureParametersDto;
+import com.sequenceiq.environment.parameter.dto.AzureResourceEncryptionParametersDto;
+import com.sequenceiq.environment.parameter.dto.AzureResourceGroupDto;
+import com.sequenceiq.environment.parameter.dto.ParametersDto;
+import com.sequenceiq.environment.parameter.dto.ResourceGroupUsagePattern;
+
+@ExtendWith(MockitoExtension.class)
+class EnvironmentEncryptionServiceTest {
+
+    private static final String CLOUD_PLATFORM = "AZURE";
+
+    @Mock
+    private CloudConnector<Object> cloudConnector;
+
+    @Mock
+    private EncryptionResources encryptionResources;
+
+    @Mock
+    private EnvironmentTagProvider environmentTagProvider;
+
+    @Mock
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Mock
+    private CredentialToCloudCredentialConverter credentialToCloudCredentialConverter;
+
+    private EnvironmentEncryptionService underTest;
+
+    @BeforeEach
+    void setUp() {
+        underTest = new EnvironmentEncryptionService(
+                credentialToCloudCredentialConverter,
+                cloudPlatformConnectors,
+                environmentTagProvider
+        );
```

Comment:
Hint. No need to instantiate this way. Just add `@InjectMocks` to `underTest` and Mockito will instantiate the class before each test method using either of three ways: constructor injection, field injection, property setter injection. The constructor will win this time, automatically receiving the correct mock instances from the above fields.